src/compiler/parse/state/tag.ts

```diff
@@ -197,6 +197,14 @@ export default function tag(parser: Parser) {
 
 	parser.eat('>', true);
 
+	// A leading newline character immediately following the pre element start tag is stripped
+	// See spec: https://www.w3.org/TR/2011/WD-html5-20110113/grouping-content.html#the-pre-element
+	if (!is_closing_tag && name === 'pre') {
+		if (!parser.eat('\r\n')) {
```

Comment:
I'm unsure if carriage return is already stripped from the source before feed into the parser?

Couldn't find any references to `CRLF` or '\r' in the parser, but I don't know.